core-java-modules/core-java-14/src/test/java/com/baeldung/java14/foreign/api/ForeignMemoryUnitTest.java

```diff
@@ -1,84 +1,78 @@
 package com.baeldung.java14.foreign.api;
 
-import jdk.incubator.foreign.*;
-import org.junit.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.nio.ByteBuffer;
 
-import static org.hamcrest.core.Is.is;
-import static org.junit.Assert.assertThat;
+import org.junit.jupiter.api.Test;
 
-import java.lang.invoke.VarHandle;
-import java.nio.ByteOrder;
+import jdk.incubator.foreign.*;
 
 public class ForeignMemoryUnitTest {
 
     @Test
-    public void whenAValueIsSet_thenAccessTheValue() {
-        long value = 10;
-        MemoryAddress memoryAddress =
-          MemorySegment.allocateNative(8).baseAddress();
-        VarHandle varHandle = MemoryHandles.varHandle(long.class,
-          ByteOrder.nativeOrder());
-        varHandle.set(memoryAddress, value);
-        assertThat(varHandle.get(memoryAddress), is(value));
+    void whenAValueIsSet_thenAccessTheValue() {
+        try (ResourceScope scope = ResourceScope.newConfinedScope()) {
+            MemorySegment segment = MemorySegment.allocateNative(4, scope);
+            ByteBuffer buffer = segment.asByteBuffer();
+
+            buffer.putInt(0, 42);
+            int value = buffer.getInt(0);
+
+            assertEquals(42, value);
+        }
     }
 
     @Test
-    public void whenMultipleValuesAreSet_thenAccessAll() {
-        VarHandle varHandle = MemoryHandles.varHandle(int.class,
-          ByteOrder.nativeOrder());
-
-        try(MemorySegment memorySegment =
-          MemorySegment.allocateNative(100)) {
-            MemoryAddress base = memorySegment.baseAddress();
-            for(int i=0; i<25; i++) {
-                varHandle.set(base.addOffset((i*4)), i);
-            }
-            for(int i=0; i<25; i++) {
-                assertThat(varHandle.get(base.addOffset((i*4))), is(i));
-            }
+    void whenMultipleValuesAreSet_thenAccessAll() {
+        try (ResourceScope scope = ResourceScope.newConfinedScope()) {
+            MemorySegment segment = MemorySegment.allocateNative(12, scope);
+            ByteBuffer buffer = segment.asByteBuffer();
+
+            buffer.putInt(0, 10);
+            buffer.putInt(4, 20);
+            buffer.putInt(8, 30);
+
+            assertEquals(10, buffer.getInt(0));
+            assertEquals(20, buffer.getInt(4));
+            assertEquals(30, buffer.getInt(8));
         }
     }
 
     @Test
```

Comment:
Hi @vunamtien 

I think we haven't replicated the original purpose and parameters of the old test method.
For instance, with ByteBuffer we allocate memory on the heap, while with MemorySegment we allocate native memory.
Moreover, in original test, we allocate memory for a sequence of 25 values, while in new test, I don't see that.

Could you give your feedback on this?